### PR TITLE
chore: Fix jest trying to run transformed test files

### LINKED ISSUE #874: Jest tests failing after `swc` builds esm files
#### Steps to reproduce
1. Run `yarn dev` in `packages/itwinui-react`
1. Then run `yarn test` in `packages/itwinui-react`

#### Actual behavior
In `packages/itwinui-react`, when running `yarn test` after running `yarn dev`, some of the jest tests fail.

image

All of the above failures give an error message similar to the one below

image

Here is the `ColorPicker.test.js` file that failed

image

#### Additional information
This is happening because jest is finding the `<component-name>.test.js` and running them. This is double work since jest also runs the `<component-name>.test.tsx` files with `ts-jest`.

#### Expected behavior
* Should not get the above error when running `yarn test`
* The `<component-name>.test.js` unit test files generated by `swc` should not run. Only the `<component-name>.test.tsx` files should run.
* We could also look into making `swc` not produce the unit test files. More info in [iTwinUI-react PR #969](https://github.com/iTwin/iTwinUI-react/pull/969) and [iTwin/iTwinUI-react@`684456b` (#969)](https://github.com/iTwin/iTwinUI-react/pull/969/commits/684456bf0953be1c582a97d1360b3559fac359a2) commit where `swc` generated the `<component-name>.test.js` units test files too.

#### Workaround

Ignore the test files in the `cjs` and `esm` folders in `jest.config.js`

image

### PR DESCRIPTION
**Pull request by mayank99**
Fixes #874 by ignoring telling jest to ignore the test files built by swc.

Also fixed formatting of jest config to remove extra newlines.

### CODE REVIEW

File: packages/itwinui-react/jest.config.js [hunk 1]
```diff
@@ -209,11 +209,11 @@ module.exports = {
...
 
-  // testPathIgnorePatterns: [
-
-  //   "\\\\node_modules\\\\"
-
-  // ],
+  testPathIgnorePatterns: [
+    '\\\\node_modules\\\\',
+    '\\\\esm\\\\',
+    '\\\\cjs\\\\',
```
Comment (rohan-kadkol) [inline packages/itwinui-react/jest.config.js:215]: Just curious, what is the reason for the backslashes? I'm guessing in L213 for example becomes `\\node_modules\\`, but not sure why the backslashes are needed.

Comment (mayank99) [reply packages/itwinui-react/jest.config.js]: i actually could not even get it to work with that. changing it to `/` works. can you confirm on windows? go inside `packages/itwinui-react` and run `yarn dev` and `yarn test` in parallel terminals

Comment (rohan-kadkol) [reply packages/itwinui-react/jest.config.js]: For me on Windows, it worked for both: `\\\\` and `/`

Comment (mayank99) [reply packages/itwinui-react/jest.config.js]: ok, lets go with `/` then

**Review by rohan-kadkol — APPROVED** [review]

**Pull request merged**